### PR TITLE
update sorting for Home Collections 'Print Labels' and 'Assign Kits'

### PR DESCRIPTION
**Pull request by JoeArmani**
Related: https://github.com/episphere/connect/issues/960
 
•Update sorting order for Home Collections 'Assign Kits'
•Use .select() to pull less data
•Update naming conventions
•Add TODOs for future enhancements (this was a late addition, and future enhancements require coordination with the Biospecimen team)

### CODE REVIEW

**Review by jhflorey — APPROVED** [review]

**Comment by jhflorey** [comment]
Looks good to me.

File: utils/firestore.js [hunk 1]
```diff
@@ -2185,10 +2202,15 @@ const queryKitsByReceivedDate = async (receivedDateTimestamp) => {
...
+        const snapshot = await db.collection("participants")
+            .select(...participantHomeCollectionKitFields)
+            .where(`${collectionDetails}.${baseline}.${bioKitMouthwash}.${kitStatus}`, '==', addressPrinted)
+            .orderBy(`${collectionDetails}.${baseline}.${bloodOrUrineCollectedTimestamp}`)
+            .get();
 
-        const snapshot = await db.collection("participants").where(`${collectionDetails}.${baseline}.${bioKitMouthwash}.${kitStatus}`, '==', fieldMapping.addressPrinted).get();
-        if (snapshot.size !== 0) return snapshot.docs.map(doc => processParticipantData(doc.data(), false));
+        if (snapshot.size !== 0) return snapshot.docs.map(doc => processParticipantHomeMouthwashKitData(doc.data(), false));
         else return false;
```
Comment (we-ai) [inline utils/firestore.js:2214]: Should it return an empty array to make data type consistent?

File: utils/firestore.js [hunk 2]
```diff
@@ -2148,26 +2148,43 @@ const checkCollectionUniqueness = async (supplyId, collectionId) => {
...
+            .where(`${activityParticipantRefusal}.${baselineMouthwashSample}`, '==', no)
+            .where(`${collectionDetails}.${baseline}.${bloodOrUrineCollected}`, '==', yes)
+            .where(`${collectionDetails}.${baseline}.${bloodOrUrineCollectedTimestamp}`, '>=', '2024-04-01T00:00:00.000Z')
+            .orderBy(`${collectionDetails}.${baseline}.${bloodOrUrineCollectedTimestamp}`)
+            .get();
+
+        return snapshot.docs.map(document => processParticipantHomeMouthwashKitData(document.data(), true));
     } catch (error) {
         console.error(error);
         return new Error(error);
```
Comment (we-ai) [inline utils/firestore.js:2190]: This returned Error instance is not handled by calling function.

**Review by we-ai — APPROVED** [review]
Looks good.
I suggest throwing an Error instance instead of returning it.

File: utils/firestore.js (hunk 1 above)
Comment (JoeArmani) [reply utils/firestore.js]: Yes, was trying to touch as little of his code as possible. Updated.

File: utils/firestore.js (hunk 2 above)
Comment (JoeArmani) [reply utils/firestore.js]: same as above...updated/handled.

File: utils/firestore.js (hunk 1 above)
Comment (we-ai) [reply utils/firestore.js]: Thanks. I usually do cleanups during implementation.

**Pull request merged**